src/main/kotlin/no/item/xp/plugin/writer/TypeScriptWriter.kt

```diff
@@ -1,3 +1,68 @@
 package no.item.xp.plugin.writer
 
-class TypeScriptWriter
+import java.io.File
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.InputType
+import no.item.xp.plugin.models.XmlType
+import java.io.BufferedWriter
+
+@Suppress("FunctionName")
+fun TypeScriptWriter(interfaceName: String, type: XmlType, objList: List<GeneratedField>?): String {
+  var content = "export interface "
+  content += xmlTypeFormatted(type) + "{\n"
+  if (objList != null) {
+    for (fields: GeneratedField in objList) {
+      content += returnCommentFromLabel(fields)
+      content += returnStringValue(fields)+"\n"
+      content += "\n"
+    }
+  }
+  content+="}"
+  File(interfaceName).bufferedWriter().use { out: BufferedWriter ->
+    out.write(content)
+  }
+  return interfaceName
+}
+
+fun returnStringValue(fields: GeneratedField): String {
+  var returnString = ""
+  returnString+=fields.name
+  returnString+= isNullable(fields.nullable)
+  returnString+=": "
+  returnString+=typeScriptType(fields.type)
+  returnString+=";"
+  return returnString
+}
+
+fun typeScriptType(type: InputType): String {
+  return when(type){
+    InputType.CHECKBOX -> "boolean"
+    InputType.DOUBLE -> "number"
+    InputType.LONG -> "number"
+    else -> "string"
+  }
+}
+
+fun isNullable(bol: Boolean):String{
+  return if (!bol) "" else "?"
+}
```

Comment:
Rename `bol` to that gives more meaning. Like `valueIsNullable`, then it can be read like "if value is nullable", and it is easier for a colleague to understand what you are trying to do.

Don't negate if you don't have to.
```suggestion
fun isNullable(valueIsNullable: Boolean): String{
  return if (valueIsNullable) "?" else ""
}
```

Reply:
chaned to valueIsNullable, and switched negation